packages/app/src/app/overmind/namespaces/live/actions.ts

```diff
@@ -187,6 +192,57 @@ export const sendCurrentSelection: Action = ({ state, effects }) => {
   }
 };
 
+export const sendCurrentViewRange: Action = ({ state, effects }) => {
+  if (!state.live.roomInfo) {
+    return;
+  }
+
+  if (state.live.isCurrentEditor) {
+    const { liveUserId, currentViewRange } = state.live;
+    if (liveUserId && currentViewRange) {
+      effects.live.sendUserViewRange(
+        state.editor.currentModuleShortid,
+        liveUserId,
+        currentViewRange
+      );
+    }
+  }
+};
+
+export const onViewRangeChanged: Action<UserViewRange> = (
+  { state, effects },
+  viewRange
+) => {
+  if (!state.live.roomInfo) {
+    return;
+  }
+
+  if (state.live.isCurrentEditor) {
+    const { liveUserId } = state.live;
+    const moduleShortid = state.editor.currentModuleShortid;
+    if (!liveUserId) {
+      return;
+    }
+
+    state.live.currentViewRange = viewRange;
+    const userIndex = state.live.roomInfo.users.findIndex(
+      u => u.id === liveUserId
+    );
+
+    if (userIndex > -1) {
```

Comment:
Since it can only be `-1` or higher, this is a bit more correct I think

```suggestion
    if (userIndex !== -1) {
```